### PR TITLE
UI update send flow

### PR DESCRIPTION
**Pull request by KVNLS**
### 📝 Description

- Update Search bar to unify UIs
- Update Native Design System with updated/new components (SearchBar + AnimatedInput + BaseInput)
- Update Send Flow with new DS components.

https://www.figma.com/file/wGzwuUVo0rkCJ3sM8cpbDY/%F0%9F%93%B1-Native-Library?node-id=14470:38382&mode=dev

https://www.figma.com/file/n5044zBaOWyNMFVYhcCZsW/Send?node-id=2954:11494&mode=dev

## Before

https://github.com/LedgerHQ/ledger-live/assets/31533861/1acb3c7e-dfd5-414e-aa62-a45519b5af8c

### Light Mode

https://github.com/LedgerHQ/ledger-live/assets/31533861/2d68c1b9-8f00-4a08-bfca-943a0aee5d53

Simulator Screenshot - iPhone 15 Pro - 2024-02-26 at 09 34 55

### Dark Mode

https://github.com/LedgerHQ/ledger-live/assets/31533861/4fa5ef9c-0286-4959-b2e8-6c060a0888e8

Simulator Screenshot - iPhone 15 Pro - 2024-02-26 at 09 34 02

### ❓ Context

- **JIRA or GitHub link**: 
- https://ledgerhq.atlassian.net/browse/LIVE-10374
- https://ledgerhq.atlassian.net/browse/LIVE-10376
- https://ledgerhq.atlassian.net/browse/LIVE-10377
- https://ledgerhq.atlassian.net/browse/LIVE-10378

### ✅ Checklist

Pull Requests must pass the CI and be code reviewed. Set as Draft if the PR is not ready.

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

---

### 🧐 Checklist for the PR Reviewers

- [ ] **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- [ ] **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- [ ] **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- [ ] **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- [ ] **Any new dependencies** have been justified and documented.
- [ ] **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by gre — DISMISSED** [review]

File: libs/ui/packages/native/src/components/cta/Button/index.tsx [hunk 1]
```diff
@@ -116,12 +118,15 @@ const ButtonContainer = (props: ButtonProps & { hide?: boolean }): React.ReactEl
...
+  const IconNode = useMemo(() => {
+    if (isNewIcon) {
+      return Icon;
+    }
+    return (
       (iconName && <IconComponent name={iconName} size={ctaIconSize[size]} color={text.color} />) ||
-      (Icon && <Icon size={ctaIconSize[size]} color={text.color} />),
-    [iconName, size, Icon, text.color],
-  );
+      (typeof Icon === "function" && <Icon size={ctaIconSize[size]} color={text.color} />)
```
Comment (hedi-edelbloute) [inline libs/ui/packages/native/src/components/cta/Button/index.tsx:127]: Why is this needed ? (and the cast below)

**Review by hedi-edelbloute — DISMISSED** [review]

**Review by sarneijim — APPROVED** [review]

**Review by haammar-ledger — COMMENTED** [review]
LGTM for blockchain-specific part :heavy_check_mark: 

Any chance some of these changes could be covered with unit tests?
I assume they are covered with e2e tests though, since they are on main parts of the UI

File: libs/ui/packages/native/src/components/cta/Button/index.tsx (hunk 1 above)
Comment (KVNLS) [reply libs/ui/packages/native/src/components/cta/Button/index.tsx:127]: Tbh I tried conditional props without success
```
type CondtionalProps =
  | { isNewIcon: true; Icon: JSX.Element }
  | { isNewIcon?: false; Icon?: IconType };

export type ButtonProps = TouchableOpacityProps &
  BaseStyledProps & {
    onPressWhenDisabled?: TouchableOpacityProps["onPress"];
    iconName?: string;
    type?: "main" | "shade" | "error" | "color" | "default";
    size?: "small" | "medium" | "large";
    iconPosition?: "right" | "left";
    outline?: boolean;
    disabled?: boolean;
    pressed?: boolean;
    children?: React.ReactNode;
    pending?: boolean;
    displayContentWhenPending?: boolean;
    buttonTestId?: string;
  } & CondtionalProps;
```
Then I have the following TS error in the Button in LLM:
```
Types of property 'isNewIcon' are incompatible.
        Type 'boolean | undefined' is not assignable to type 'false | undefined'.
```
I will try to see if I can't find another fix.

File: .changeset/big-ears-suffer.md
[the remark this answers is not in the document]
Comment (KVNLS) [reply .changeset/big-ears-suffer.md:1]: Nice catch, thanks!

**Comment by KVNLS** [comment]
> LGTM for blockchain-specific part ✅
> 
> Any chance some of these changes could be covered with unit tests? I assume they are covered with e2e tests though, since they are on main parts of the UI

As it's only a design change + Rework of DS components I have not changed the tests we have.
Indeed it's covered by e2e tests but what we should do next are:
- Add testing tools on our DS system to unit test our DS components
- Migrate the send flow to new arch and write Integration tests

**Review by sarneijim — APPROVED** [review]

**Review by haammar-ledger — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

File: libs/ui/packages/native/src/components/cta/Button/index.tsx (hunk 1 above)
Comment (hedi-edelbloute) [reply libs/ui/packages/native/src/components/cta/Button/index.tsx:127]: Discussed in DMs

File: apps/ledger-live-mobile/src/screens/SendFunds/03a-AmountCoin.tsx [hunk 2]
```diff
@@ -122,6 +119,7 @@ export default function SendAmountCoin({ navigation, route }: Props) {
     setTransaction(bridge.updateTransaction(transaction, {}));
   }, [setTransaction, account, parentAccount, transaction]);
   const blur = useCallback(() => Keyboard.dismiss(), []);
+  const onMaxSpendableLearnMore = useCallback(() => Linking.openURL(urls.maxSpendable), []);
```
Comment (mcayuelas-ledger) [inline apps/ledger-live-mobile/src/screens/SendFunds/03a-AmountCoin.tsx:122]: Same

File: apps/ledger-live-mobile/src/families/solana/DelegationFlow/SelectAmount.tsx [hunk 3]
```diff
@@ -124,6 +124,7 @@ export default function DelegationSelectAmount({ navigation, route }: Props) {
   const { amount } = status;
   const unit = getAccountUnit(account);
   const currency = getAccountCurrency(account);
+  const onMaxSpendableLearnMore = useCallback(() => Linking.openURL(urls.maxSpendable), []);
```
[the remark this answers is not in the document]
Comment (KVNLS) [reply apps/ledger-live-mobile/src/families/solana/DelegationFlow/SelectAmount.tsx:127]: `useLocalizedUrl` is only available on LLD, am I wrong?

[the remark this answers is not in the document]
Comment (mcayuelas-ledger) [reply apps/ledger-live-mobile/src/families/solana/DelegationFlow/SelectAmount.tsx:127]: Oh yes :'( my bad, we need to implement this on LLM

**Review by sshmaxime — APPROVED** [review]
LGTM !
